rtflib/rtf.py

```diff
@@ -81,30 +94,29 @@ def __init__(self):
         self.preelements: list[RtfElement] = []
         self.elements: list[RtfElement] = []
         self.colors = []
+        self.color_counter = 1  # 0 is reserved
 
     def addstrict(self):
         self.elements.append(RtfCode("\\f0\\fs24\\cf0\n"))
 
+    def add_color(self, red: int, green: int, blue: int) -> int:
+        for c in self.colors:
+            if c.red == red and c.green == green and c.blue == blue:
+                return c
+        # no match found, so append
+        c = Color(red, green, blue, self.color_counter)
+        self.colors.append(c)
+        self.color_counter += 1
+        return c
+        
     def add(self, element: RtfElement) -> None:
         if element.iscompatible("rtf"):
             self.elements.append(element)
             if "color" in dir(element) and element.color:
```

Comment:
It would be cleaner to get rid of this and update the examples so that the user should do:
```python
line = Line("hello world", color=rtf.add_color(255, 0, 0))
```

Also, replace `"color" in dir(element)` with `hasattr(element, "color")` (although this might get removed)

Reply:
ok. so if the color does not already exist then inline the rtf.add_color() and rely on `add_color()` to return the old color if this new one already exists ?